File: application/backend-credit-card-enrollment/backend-c#/CreditCardEnrollment/CreditCard/Enrollment/Projection/EnrollmentProjectionController.cs
using CreditCardEnrollment.Common.Ambar;
using CreditCardEnrollment.Common.Projection;
using CreditCardEnrollment.Common.SerializedEvent;
using CreditCardEnrollment.CreditCard.Enrollment.Projection.EnrollmentList;
using CreditCardEnrollment.CreditCard.Enrollment.Projection.IsProductActive;
using Microsoft.AspNetCore.Mvc;

namespace CreditCardEnrollment.CreditCard.Enrollment.Projection;

[ApiController]
[Route("api/v1/credit_card/enrollment/projection")]
[Produces("application/json")]
[Consumes("application/json")]
public class EnrollmentProjectionController : ProjectionController {
    private readonly IsProductActiveProjectionHandler _isProductActiveProjectionHandler;
    private readonly EnrollmentListProjectionHandler _enrollmentListProjectionHandler;

    public EnrollmentProjectionController(
        MongoTransactionalProjectionOperator mongoOperator,
        Deserializer deserializer,
        IsProductActiveProjectionHandler isProductActiveProjectionHandler,
        EnrollmentListProjectionHandler enrollmentListProjectionHandler)
        : base(mongoOperator, deserializer) {
        _isProductActiveProjectionHandler = isProductActiveProjectionHandler;
        _enrollmentListProjectionHandler = enrollmentListProjectionHandler;
    }

    [HttpPost("is_card_product_active")]
    public IActionResult ProjectIsCardProductActive([FromBody] AmbarHttpRequest request) {
        return new ContentResult {
            Content = ProcessProjectionHttpRequest(
                request, 
                _isProductActiveProjectionHandler, 
                "CreditCard_Enrollment_IsProductActive"),
            ContentType = "application/json",
            StatusCode = 200
        };
    }

    [HttpPost("enrollment_list")]
    public IActionResult ProjectEnrollmentList([FromBody] AmbarHttpRequest request) {
        return new ContentResult {
            Content = ProcessProjectionHttpRequest(
                request, 
                _enrollmentListProjectionHandler, 
                "CreditCard_Enrollment_EnrollmentList"),
            ContentType = "application/json",
            StatusCode = 200
        };
    }
}

File: application/backend-credit-card-enrollment/backend-typescript/src/creditCard/enrollment/projection/EnrollmentProjectionController.ts
import { Request, Response, Router } from 'express';
import { MongoTransactionalProjectionOperator } from '../../../common/projection/MongoTransactionalProjectionOperator';
import { Deserializer } from '../../../common/serializedEvent/Deserializer';
import { AmbarHttpRequest } from '../../../common/ambar/AmbarHttpRequest';
import {ProjectionController} from "../../../common/projection/ProjectionController";
import {IsProductActiveProjectionHandler} from "./isProductActive/IsProductActiveProjectionHandler";
import {EnrollmentListProjectionHandler} from "./enrollmentList/EnrollmentListProjectionHandler";

export class EnrollmentProjectionController extends ProjectionController{
    public readonly router: Router;

    constructor(
        mongoOperator: MongoTransactionalProjectionOperator,
        deserializer: Deserializer,
        private readonly isProductActiveProjectionHandler: IsProductActiveProjectionHandler,
        private readonly enrollmentListProjectionHandler: EnrollmentListProjectionHandler
    ) {
        super(mongoOperator, deserializer);
        this.router = Router();
        this.router.post('/is_card_product_active', this.projectIsCardProductActive.bind(this));
        this.router.post('/enrollment_list', this.projectEnrollmentList.bind(this));
    }

    private async projectIsCardProductActive(req: Request, res: Response): Promise<void> {
        const response = await this.processProjectionHttpRequest(
            req.body as AmbarHttpRequest,
            this.isProductActiveProjectionHandler,
            'CreditCard_Enrollment_IsProductActive'
        );
        res.status(200).contentType('application/json').send(response);
    }

    private async projectEnrollmentList(req: Request, res: Response): Promise<void> {
        const response = await this.processProjectionHttpRequest(
            req.body as AmbarHttpRequest,
            this.enrollmentListProjectionHandler,
            'CreditCard_Enrollment_EnrollmentList'
        );
        res.status(200).contentType('application/json').send(response);
    }
}
